### ISSUE TITLE
TripleA HotSpots

### ISSUE THREAD

**Issue by RoiEXLab** [triplea-game/triplea#2948]
I did some basic profiling to identify potential thresholds in the TripleA codebase inspired by [this post](https://forums.triplea-game.org/topic/552/triplea-optimized-jvm-switches).
The results are really not suprising, but I though I want to share them nevertheless.
I recorded this data during a game on WAW with all players set to hard AI.
Memory:
![Memory](https://user-images.githubusercontent.com/8350879/35767079-184ae40e-08e5-11e8-81cc-6d7e118c50ba.png)
CPU:
![CPU](https://user-images.githubusercontent.com/8350879/35767092-670cdcf0-08e5-11e8-8561-ad401560a42d.png)

So in short: The UI heavily relies on java.awt.Rectangles, OddsCalculator and RouteFinder take much time (not in this particular screenshot, but when planning the move etc.), Deserialization/IO Operations for GameData as well, Reflective calls for Attachments do take much time as well.
So we should definitely keep that in mind when doing a breaking release.

I'll close this issue when everyone has seen it. Please remove your assignment once you read this issue.

**Comment by ssoloff** [comment]
:+1:

**Comment by RoiEXLab** [comment]
Closing this
@DanVanAtta
